Cargo.toml

```diff
@@ -87,16 +87,64 @@ rust.unreachable_pub = "warn"
 rustdoc.all = "warn"
 rust.unused_must_use = "deny"
 rust.rust_2018_idioms = "deny"
-clippy.empty_line_after_outer_attr = "deny"
-clippy.derive_partial_eq_without_eq = "deny"
-clippy.redundant_clone = "deny"
-clippy.trait_duplication_in_bounds = "deny"
-clippy.uninlined_format_args = "deny"
-clippy.equatable_if_let = "deny"
-clippy.or_fun_call = "deny"
-clippy.branches_sharing_code = "deny"
-clippy.useless_let_if_seq = "deny"
-clippy.large_stack_frames = "deny"
+
+[workspace.lints.clippy]
+# These are some of clippy's nursery (i.e., experimental) lints that we like.
+# By default, nursery lints are allowed. Some of the lints below have made good
+# suggestions which we fixed. The others didn't have any findings, so we can
+# assume they don't have that many false positives. Let's enable them to
+# prevent future problems.
+branches_sharing_code = "deny"
+clear_with_drain = "deny"
+derive_partial_eq_without_eq = "deny"
+empty_line_after_outer_attr = "deny"
+equatable_if_let = "deny"
+imprecise_flops = "deny"
+iter_on_empty_collections = "deny"
+iter_with_drain = "deny"
+large_stack_frames = "deny"
+manual_clamp = "deny"
+mutex_integer = "deny"
+needless_pass_by_ref_mut = "deny"
+nonstandard_macro_braces = "deny"
+or_fun_call = "deny"
+path_buf_push_overwrite = "deny"
+read_zero_byte_vec = "deny"
+redundant_clone = "deny"
+suboptimal_flops = "deny"
+suspicious_operation_groupings = "deny"
+trailing_empty_array = "deny"
+trait_duplication_in_bounds = "deny"
+transmute_undefined_repr = "deny"
+trivial_regex = "deny"
+tuple_array_conversions = "deny"
+uninhabited_references = "deny"
+unused_peekable = "deny"
+unused_rounding = "deny"
+useless_let_if_seq = "deny"
```

Comment:
Not a big fan of denying lints, and we deny them in clippy CI anyway.

Reply:
What do you mean? I don't think I understand.

Reply:
Use "warn" instead, as we `--deny warnings` in CI to catch them.

Reply:
Oh gotcha, yeah that's fair. Changed them to warn.